### ISSUE TITLE
[drum-sampler] crash when moving the inpoint

### ISSUE THREAD

**Issue by cbsrobot** [topisani/OTTO#28]
with commit 3573ea3d6 the sampler crashes when using A ans Q Keys
in the debugger it looks like p2 has no values.

```
Process 8232 stopped
thread #1: tid = 0x26a16, 0x000000010009a648 otto`void otto::ui::vg::Canvas::plotRounded<std::__1::__wrap_iter<otto::ui::vg::Point const*> >(this=0x00007fff5fbfe4c0, first=__wrap_iter<const otto::ui::vg::Point *> @ 0x00007fff5fbfc6f0, last=__wrap_iter<const otto::ui::vg::Point *> @ 0x00007fff5fbfc6e8, max_radius=2.97855616) + 728 at canvas.hpp:435, queue = 'com.apple.main-thread', stop reason = EXC_BAD_ACCESS (code=2, address=0x101400000)
    frame #0: 0x000000010009a648 otto`void otto::ui::vg::Canvas::plotRounded<std::__1::__wrap_iter<otto::ui::vg::Point const*> >(this=0x00007fff5fbfe4c0, first=__wrap_iter<const otto::ui::vg::Point *> @ 0x00007fff5fbfc6f0, last=__wrap_iter<const otto::ui::vg::Point *> @ 0x00007fff5fbfc6e8, max_radius=2.97855616) + 728 at canvas.hpp:435
   432           moveTo(*first);
   433           for (auto [p1, p2] : util::adjacent_pairs(first, last))
   434           {
-> 435             float dx = std::abs(p2.x - p1.x);
   436             float dy = std::abs(p2.y - p1.y);
   437             float r = std::min({dx / 2.f, dy / 2.f, max_radius});
   438             vg::Point md = (p1 + p2) / 2.0;
```

**Comment by topisani** [comment]
fairly certain this is fixed with 5040b5dc83732eea71eae3b23a1cd5aa8b6ae688 (currenty only on the `engine` branch). could you try it out @cbsrobot?

**Comment by topisani** [comment]
sampler has been removed - see #66
